### PR TITLE
Update the API catalog format and code support

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

**Comment by jskeet** [comment]
Will investigate the CI failures

**Comment by jskeet** [comment]
Ah, whoops - I'd forgotten that the python scripts need updating. Will add a commit with that.

File: tools/Google.Cloud.Tools.ProjectGenerator/Program.cs [hunk 1]
```diff
@@ -131,15 +131,15 @@ static int Main()
...
-                Console.WriteLine($"API catalog contains {apis.Count} entries");
+                var catalog = ApiCatalog.Load();
+                Console.WriteLine($"API catalog contains {catalog.Apis.Count} entries");
                 // Now we know we can parse the API catalog, let's reformat it.
-                ReformatApiCatalog();
-                RewriteReadme(apis);
-                RewriteDocsRootIndex(apis);
-                HashSet<string> apiNames = new HashSet<string>(apis.Select(api => api.Id));
+                ReformatApiCatalog(catalog);
+                RewriteReadme(catalog.Apis);
```
Comment (amanda-tarafa) [inline tools/Google.Cloud.Tools.ProjectGenerator/Program.cs:138]: nit: consider passing the catalog here and in the next one, even if you can do with the list here.

File: tools/Google.Cloud.Tools.ProjectGenerator/Program.cs [hunk 2]
```diff
@@ -131,15 +131,15 @@ static int Main()
...
+                RewriteDocsRootIndex(catalog.Apis);
+                HashSet<string> apiNames = new HashSet<string>(catalog.Apis.Select(api => api.Id));
```
Comment (amanda-tarafa) [inline tools/Google.Cloud.Tools.ProjectGenerator/Program.cs:140]: nit: And consider moving this to the catalog.

File: tools/Google.Cloud.Tools.ReleaseManager/Program.cs [hunk 3]
```diff
@@ -229,10 +211,10 @@ private static void ShowVersions(string[] args)
 
         private static List<ApiVersionPair> FindChangedVersions()
         {
-            var currentCatalog = ApiMetadata.LoadApis();
+            var currentCatalog = ApiCatalog.Load();
             var masterCatalog = LoadMasterCatalog();
-            var currentVersions = currentCatalog.ToDictionary(api => api.Id, api => api.Version);
-            var masterVersions = masterCatalog.ToDictionary(api => api.Id, api => api.Version);
+            var currentVersions = currentCatalog.Apis.ToDictionary(api => api.Id, api => api.Version);
```
Comment (amanda-tarafa) [inline tools/Google.Cloud.Tools.ReleaseManager/Program.cs:216]: nit: Consider moving to the catalog.

**Comment by jskeet** [comment]
Nice - all your suggestions are things I'd been considering :) Will implement them tomorrow in a new commit and merge.

**Pull request merged**